### DISCUSSION TITLE
how to use next font google in a server with blocked internet access

### DISCUSSION THREAD

**Discussion by alexmondaini** [vercel/next.js#65595] [Help]
### Summary

I am new to the javascript world and next.js , and my environment blocks access to google apis which are accessed when a new font from googe is imported. I am following this tutorial from next.js https://nextjs.org/learn/dashboard-app/optimizing-fonts-images and the idea is to set my app/layout.tsx like that:

```
import '@/app/ui/global.css';
import { inter } from '@/app/ui/fonts';
 
export default function RootLayout({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <html lang="en">
      <body className={`${inter.className} antialiased`}>{children}</body>
    </html>
  );
}
```

and app/ui/fonts like this:

```
import { Inter } from 'next/font/google';
 
export const inter = Inter({ subsets: ['latin'] });
```

The problem is that download step on my server will not work, and then another font is used as a fallback. I would like to use fonts from google in my project. What is the way to go in that scenario ?

Thanks

**Comment by zanzlender** [comment]
You can download the fonts, then add them to the public folder and import/use those as local fonts. Or you can add them anywhere and just import them, this might make your bundles a bit bigger but you won't need to download the from Google API.

**Reply by DiskCrasher** [reply]
This is really not an acceptable long-term solution since you'll have to manually ensure the font files in your project are kept up-to-date.

**Reply by BenSluis** [reply]
> You can download the fonts, then add them to the public folder and import/use those as local fonts. Or you can add them anywhere and just import them, this might make your bundles a bit bigger but you won't need to download the from Google API.

How can I do that?

**Comment by alexmondaini** [comment]
ok, did that but noticed that while the page is loading there is a shift between one font and the other, is this normal ?

**Reply by nayounsang** [reply]
This is the "font flash". You can use <link rel="preload"> or font file optimization or other optimization.

**Reply by alexmondaini** [reply]
I do not understand how to do this, could you send me some link of how font optimization is done in next js ?
